iris_grib/_save_rules.py

```diff
@@ -670,20 +683,81 @@ def grid_definition_template_30(cube, grib):
     )
     eccodes.codes_set(grib, "LaD", cs.central_lat / _DEFAULT_DEGREES_UNITS)
     eccodes.codes_set(grib, "LoV", central_lon / _DEFAULT_DEGREES_UNITS)
+
+
+def grid_definition_template_20(cube, grib):
+    """
+    Set keys within the provided GRIB message based on
+    Grid Definition Template 3.20.
+
+    Template 3.20 is used to represent a Polar Stereographic grid.
+
+    """
+    eccodes.codes_set(grib, "gridDefinitionTemplateNumber", 20)
+
+    # Retrieve the cube coord system.
+    cs = cube.coord(dimensions=[0]).coord_system
+
+    _perspective_projection_common(cube, grib)
+
+    # Is this a north or south polar stereographic projection?
+    if cs.central_lat == -90.0:
+        centre_flag = 0x80
+    elif cs.central_lat == 90.0:
+        centre_flag = 0x0
+    else:
+        emsg = (
+            "Bipolar and symmetric polar stereo projections "
```

Comment:
Where do you get the class descriptions for this ?
I think it would be better to describe a case with central_lat not +/-90 just as "non-polar stereographic projections".  But perhaps there are other subtleties here that I'm not aware of ?

Reply:
[ECMWF | Section 3 - Template 20 : Polar stereographic projection](https://codes.ecmwf.int/grib/format/grib2/templates/3/20/)

>| Octets | Key | Type | Content |
>| --- | --- | --- | --- |
>| ... | ... | ... | ... |
>| 64 | projectionCentreFlag | codeflag | Projection centre flag (See Flag Table 3.5) |
>| ... | ... | ... | ... |
>
>Notes:
>...
>( 4) Bit 2 of the projection flag is not applicable to the polar stereographic projection.

[ECMWF | Section 3 - Flag Table 5 : Projection Centre](https://codes.ecmwf.int/grib/format/grib2/ftables/3/5/)

>| Bit | Value | Meaning |
>| --- | --- | --- |
>| ... | ... | ... |
>| 2 | 1 | Projection is bi-polar and symmetric |

Reply:
What "only one projection centre is used" I really don't understand.  I guess a polar projection always has a single centre, but it might not be at either the North or South pole.  So then "LaD" would be something other than +/-90, which we are not supporting as we aren't sure if + how it is legal.
All this is just guesswork, really.

Reply:
f793bdd5711c681f937cf0844590c36c0f05b58f

Reply:
Great, I'll happily  accept that.
Thanks for your patience !